app/src/main/java/com/android/streetworkapp/model/parks/OverpassParkLocationRepository.kt

```diff
@@ -0,0 +1,85 @@
+package com.android.streetworkapp.model.parks
+
+import java.io.IOException
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.json.JSONObject
+
+class OverpassParkLocationRepository(private val client: OkHttpClient) : ParkLocationRepository {
+  /**
+   * Find all nearby Street Workout parks, from OpenStreetMap using Overpass API
+   *
+   * @param lat : latitude (Double) must be between [-90,90]
+   * @param lon : longitude (Double) must be between [-180,180]
+   * @param onSuccess : to handle successful cases
+   * @param onFailure : to handle unsuccessful cases
+   */
+  override fun search(
+      lat: Double,
+      lon: Double,
+      onSuccess: (List<ParkLocation>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    if (lat > 90 || lat < -90 || lon > 180 || lon < -180) {
```

Comment:
Suggestion: For more precise exception handling, it would be better to use two separate conditions:

```kotlin
if (lat > 90 || lat < -90) {...}
if (lon > 180 || lon < -180) {...}
```
If you split the conditions, avoid nesting them as this would increase the [cognitive complexity](https://www.sonarsource.com/blog/5-clean-code-tips-for-reducing-cognitive-complexity/) as reported by Sonar.

Reply:
I fixed it. Thank you!